lib/client/api.go

```diff
@@ -2021,3 +2071,39 @@ func ParsePortForwardSpec(spec []string) (ports ForwardedPorts, err error) {
 	}
 	return ports, nil
 }
+
+// ToStringSpec returns the same string spec which can be parsed by
+// ParseDynamicPortForwardSpec.
+func (fp DynamicForwardedPorts) ToStringSpec() (retval []string) {
+	for _, p := range fp {
+		retval = append(retval, p.ToString())
+	}
+	return retval
+}
+
+// ParseDynamicPortForwardSpec parses parameter to -D flag, i.e. strings like "remote.host:3000"
+// The opposite of this function (spec generation) is DynamicForwardPorts.ToStringSpec.
+func ParseDynamicPortForwardSpec(spec []string) (ports DynamicForwardedPorts, err error) {
```

Comment:
let's not use named return values in go in larger functions without a good reason:

https://github.com/golang/go/wiki/CodeReviewComments#named-result-parameters

> Naked returns are okay if the function is a handful of lines. Once it's a medium sized function, be explicit with your return values. Corollary: it's not worth it to name result parameters just because it enables you to use named returns. Clarity of docs is always more important than saving a line or two in your function.